src/canvas/StaticCanvas.ts

```diff
@@ -353,21 +353,7 @@ export class StaticCanvas<
    * @param {Array} vpt a Canvas 2D API transform matrix
    */
   setViewportTransform(vpt: TMat2D) {
-    const backgroundObject = this.backgroundImage,
-      overlayObject = this.overlayImage,
-      len = this._objects.length;
-
-    this.viewportTransform = vpt;
-    for (let i = 0; i < len; i++) {
-      const object = this._objects[i];
-      object.group || object.setCoords();
-    }
-    if (backgroundObject) {
-      backgroundObject.setCoords();
-    }
-    if (overlayObject) {
-      overlayObject.setCoords();
-    }
```

Comment:
why deleting those? control coords need to be set  when we change the vpt.

Reply:
We need control coords only for the active object

Reply:
And regarding corner coords, we need them on these objects only if they do not ignore the vpt

Reply:
Looking at the code we do not even check if these objects are on screen and these objects are not selectable apart from imperatively calling `setActiveObject` (which I believe should call `setCoords` if it doesn't) thus making the `setCoords` call redundant

Reply:
Exactly, the active object calls setCoords, the rest should not. Mind that `setActiveObject` call setCoords so it seals the gap.